src/components/map-controllers/PrecipMapController/PrecipMapController.js

```diff
@@ -0,0 +1,293 @@
+    updateLayerTime.bind(this, 'raster', timeIdx)();
+    updateLayerTime.bind(this, 'annotated', annotatedIndex)();
+  }
+
+  // Support functions for event/callback handlers
+
+  // TODO: split loadmap into helpers?
+  // TODO: https://github.com/pacificclimate/climate-explorer-frontend/issues/125
+  loadMap(
+    props,
+    dataSpec,
+    newVariable = false,
+  ) {
+    // Update state with all the information needed to display
+    // maps for the specified dataspec.
+    // A 'dataSpec' represents a combination of a specific variable, 
+    // emissions scenario, model, period, and run. The variable, emissions, 
+    // and model are selected by the user in a top level component, and implicitly
+    // encoded in meta and comparandMeta, which are filtered to only relevant 
+    // datasets by this component's parent.
+    // The start date, end date, and run and selected by this component - either 
+    // defaults or user selection - and supplied as the "dataSpec" variable.
+    // The data described by a dataspec may be spread across up to three data files
+    // (yearly, seasonal, monthly); the specific file needed to map a particular
+    // timestamp is determined at render time and passed to the viewer component.
+
+    const { start_date, end_date, ensemble_member } = dataSpec;
+    
+    const rasterScalarParams = scalarParams.bind(null, props.variable_id);
+    const rasterParamsPromise = getTimeParametersPromise(dataSpec, props.meta)
+      .then(rasterScalarParams)
+      .then(selectRasterPalette);
+    
+    const annotatedScalarParams = scalarParams.bind(null, props.comparand_id);
+    const annotatedParamsPromise = getTimeParametersPromise(dataSpec, props.comparandMeta)
+      .then(annotatedScalarParams);
+    
+    Promise.all([rasterParamsPromise, annotatedParamsPromise]).then(params => {
+      
+      let rasterParams = _.findWhere(params, {variableId: props.variable_id});
+      let annotatedParams = _.findWhere(params, {variableId: props.comparand_id});
+      
+      // if the variable has changed, go back to the default palette and logscale,
+      // otherwise use the previous (user-selected) values in state.
+      if(newVariable) {
+        rasterParams.palette = 'seq-Blues';
+        rasterParams.logscale = "true";
+      }
+      else {
+        rasterParams.palette = this.state.raster.palette;
+        rasterParams.logscale = this.state.raster.logscale;
+      }
+      
+      this.setState(prevState => ({
+        run: ensemble_member,
+        start_date,
+        end_date,
+        raster: rasterParams,
+        annotated: annotatedParams
+      }));
+    });  
+  }
+
+  // Handlers for dataSpec change
+
+  // TODO: https://github.com/pacificclimate/climate-explorer-frontend/issues/118
+  updateDataSpec = (encodedDataSpec) => {
+    this.loadMap(this.props, JSON.parse(encodedDataSpec));
+  };
+
+  // Handlers for settings changes
+
+  handleChangeTime = this.updateTime.bind(this);  
```

Comment:
Simplify: Declare `updateTime` as an arrow function and rename it `handleChangeTime`.
Anywhere you are binding only to `this`, just use an arrow function instead, which does the binding for you.